src/mpid/ch4/src/ch4r_rma.h

```diff
@@ -40,13 +39,10 @@ static inline int MPIDI_CH4I_do_put(const void *origin_addr,
 
     MPIDI_CH4U_EPOCH_CHECK_SYNC(win, mpi_errno, goto fn_fail);
 
-    sreq = MPIDI_CH4I_am_request_create(MPIR_REQUEST_KIND__UNDEFINED, 1);
+    /* two ref counts for progress engine and caller. */
+    sreq = MPIDI_CH4I_am_request_create(MPIR_REQUEST_KIND__UNDEFINED, 2);
```

Comment:
We shouldn't create a request of type UNDEFINED and immediately set it to type RMA.